lib/Dependency.js

```diff
@@ -4,15 +4,24 @@
 */
 "use strict";
 
+/** @typedef {import("./Module")} Module */
+/** @typedef {import("crypto").Hash} Hash */
+/** @typedef {import("acorn").SourceLocation} SourceLocation */
+
 const compareLocations = require("./compareLocations");
 const DependencyReference = require("./dependencies/DependencyReference");
 
 class Dependency {
 	constructor() {
+		/** @type {Module|null} */
 		this.module = null;
 		this.weak = false;
 		this.optional = false;
+		/** @type {(SourceLocation|string|HarmonyDependencySourceLocation)=} */
```

Comment:
Make it only `DependencyLocation`. Rewrite the cases where this is a `string` to also use `DependencyLocation`. Add a `name: string` property to the `DependencyLocation` to cover these cases.